src/main/java/com/baeldung/web/controller/OldRegistrationController.java

```diff
@@ -27,18 +27,17 @@
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.core.userdetails.UserDetailsService;
-import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RestController;
 import org.springframework.ui.Model;
 import org.springframework.validation.Errors;
 import org.springframework.web.bind.annotation.ModelAttribute;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.servlet.ModelAndView;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PostMapping;
 
-@Controller
+@RestController
```

Comment:
is this change needed here, since the mapping in this controller don't have a ResponseBody annotation, and return views instead?

Reply:
Technically no, I've just changed it for the sake of consistency. Let me know if I should revert this  change. Thanks

Reply:
Let's leave it as `@Controller`. `@RestController` is for controllers that return data.